var/spack/repos/builtin/packages/relion/package.py

```diff
@@ -65,14 +82,19 @@ def cmake_args(self):
             '-DDoublePrec_GPU=%s' % ('+double-gpu' in self.spec),
         ]
 
-        carch = self.spec.variants['cuda_arch'].value
-
         if '+cuda' in self.spec:
-            args += [
-                '-DCUDA=on',
-            ]
-            if carch is not None:
-                args += [
-                    '-DCUDA_ARCH=%s' % (carch),
-                ]
+            # relion+cuda requires selecting cuda_arch
+            if not carch:
+                # below does not work so raising an exception
+                conflicts('%gcc@4.0:6.99', msg='you must select cuda_arch')
```

Comment:
This won't work. `conflicts` can only be used at the class-level, it can't be used inside of a function.

There isn't a "not" in the spec syntax, and I don't know of a way to say `if cuda_arch=''`. You'll just have to raise a ValueError here with this message.

Reply:
Thanks

If the user does not define cuda_arch=value at command line then `carch` isn't defined so then the `if not` does indeed work.